app/components/dialog/components/allPublications/index.tsx

```diff
@@ -87,7 +90,7 @@ class AllPublicationsDialog extends React.PureComponent<AllPublicationsDialogPro
               disabled={isLoading}
               isLoading={isLoading}
               gaCategory="AllPublications"
-              content="Add Publications"
+              content={"Add " + selectedPapers.length + " Publications"}
```

Comment:
content={"Add " + selectedPapers.length + " Publications"}
You can use string interporlation.
`content={`Add ${selectedPapers.length} Publications`}`